### ISSUE TITLE
All parameters are always nullable

### ISSUE THREAD

**Issue by shocklateboy92** [epam-cross-platform-lab/swagger-dart-code-generator#747]
Sorry if this is already answered in your documentation somewhere, but I could not find it.

I noticed that the parameters to the generated chopper methods are always nullable.
```dart
  Future<chopper.Response<Genre>> GetGenreById({
    required String? id,
    String? someOtherProp,
    required String? someRequiredProp,
  }) {
```

And looking at the code, I see an unconditional call to `makeNullable()`
https://github.com/epam-cross-platform-lab/swagger-dart-code-generator/blob/9533c64536a018f5da9217ea838fbc43a5e83a8b/lib/src/code_generators/swagger_requests_generator.dart#L796

I was quite confused by this. Shouldn't we use the value of the `nullable` property from the OpenApi spec to determine this?
https://swagger.io/docs/specification/describing-parameters/#nullable
![image](https://github.com/epam-cross-platform-lab/swagger-dart-code-generator/assets/856493/33b7d43e-48b3-4174-8f46-53a4da3ff35f)

**Comment by msuddaby** [comment]
Also experiencing the same behaviour.

**Comment by AngeloAvv** [comment]
Any clarifications on this? Is this something wanted or simply a bug?

**Comment by Vovanella95** [comment]
Hi all! I think it's a bug. Will try to fix it soon!

**Comment by Vovanella95** [comment]
Hi @AngeloAvv  @msuddaby !

Current behavior is next. Property is nullable if it has `nullable: true` or it absent in 'required' list.

For example:

1st case. Name will be not nullable
```json
"Pet": {
      "type": "object",
      "required": [
        "name"
      ],
      "properties": {
        "name": {
          "type": "string",
          "example": "doggie"
        },
}
```

2nd case. Name will be nullable

```json
"Pet": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string",
          "example": "doggie"
        },
}
```

3rd case. Name will be required and nullable
```json
"Pet": {
      "type": "object",
      "required": [
        "name"
      ],
      "properties": {
        "name": {
          "type": "string",
          "example": "doggie",
          "required": true,
        },
}
```

So, it's expected behavior. That's how it meant to be.

**Comment by nmtitov** [comment]
> 2nd case. Name will be nullable


@Vovanella95 is it possible to configure generator somehow so that it doesn't check "required" list? My swagger files only use `nullable: true` when needed and I don't have control over them. So I also have non-nullable properties nullable in generated code.

**Comment by nmtitov** [comment]
I have wrote a script which iterate over json schemas in spec and appends all "required" fields to "required" list.
It rewrites json spec in place so use vcs.

https://github.com/nmtitov/rewrite-epam-swagger

Also I think this behaviour is an error. `Required` in spec means that properties should always exist in the output. It doesn't mean that such properties are nullable or not.

Reference: 
https://stackoverflow.com/a/45577763

```
In other words, required means "property must be present", regardless of its value. The type, format, etc. of the property value are separate constraints that are evaluated separately from required, but together as a combined schema.
```

/cc @Vovanella95
